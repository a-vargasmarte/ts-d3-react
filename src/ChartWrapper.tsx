import * as React from "react";
import { Component } from "react";
import D3Chart from "./D3Chart";

interface IState {
  chart: any;
}

interface IProps {
  gender: string;
}
export default class ChartWrapper extends Component<IProps, IState> {
  componentDidMount() {
    let chart: any;
    chart = new D3Chart(this.refs.chart);

    console.log(chart.update);

    this.setState({
      chart: chart
    });
    console.log(this.state);
  }

  componentWillReceiveProps(props: any) {
    console.log(this.state.chart.update);
    // let { chart } = this.state;
    this.state.chart.update(props.gender);
  }
  render() {
    return <div ref="chart"></div>;
  }
}
