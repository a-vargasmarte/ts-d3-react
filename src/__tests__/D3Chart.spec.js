import React from 'react';
import {create} from 'react-test-renderer';
import ChartWrapper from '../ChartWrapper';


describe("ChartWrapper component", () => {
    test("Component gets rendered", () => {
        const chartWrapper = create(<ChartWrapper />);
        expect(chartWrapper.toJSON()).toMatchSnapshot();
    })
})
  