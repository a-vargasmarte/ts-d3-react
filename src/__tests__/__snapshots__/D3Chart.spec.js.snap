// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartWrapper component Component gets rendered 1`] = `<div />`;
